lib/Sema/CSSimplify.cpp

```diff
@@ -1548,8 +1548,8 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
     if (!shouldAttemptFixes())
       return getTypeMatchFailure(locator);
 
-    auto *fix = MarkExplicitlyEscaping::create(
-        *this, getConstraintLocator(locator), func2);
+    auto *fix = MarkExplicitlyEscaping::create(*this, func1, func2,
```

Comment:
Nice!
That's exactly the diagnostic which was failing when I try to use the source locator as the binding locator, I definitely was missing something 😅